modules/misc/gtk.nix

```
@@ -166,14 +166,30 @@ in {
     home.packages = optionalPackage cfg.font ++ optionalPackage cfg.theme
       ++ optionalPackage cfg.iconTheme;
 
-    home.file.${cfg2.configLocation}.text =
-      concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
-      + cfg2.extraConfig;
+    home.file.${cfg2.configLocation} = {
+      text = concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
+        + cfg2.extraConfig;
+      onChange = let xsettingsd = "${pkgs.xsettingsd}/bin/xsettingsd";
+      in ''
+        DISPLAY=":0" $DRY_RUN_CMD ${xsettingsd} -c \
+        <( echo "Net/ThemeName \"${cfg.theme.name}\"" ) & \
+        ${pkgs.procps}/bin/pkill -f ${xsettingsd} || true
+      '';
+    };
 
     home.sessionVariables.GTK2_RC_FILES = cfg2.configLocation;
 
-    xdg.configFile."gtk-3.0/settings.ini".text =
-      toGtk3Ini { Settings = ini // cfg3.extraConfig; };
+    xdg.configFile."gtk-3.0/settings.ini" = {
+      text = toGtk3Ini { Settings = ini // cfg3.extraConfig; };
+      onChange = let
+        gsettings-schemas =
+          "${pkgs.gsettings-desktop-schemas}/share/gsettings-schemas/gsettings-desktop-schemas-${pkgs.gsettings-desktop-schemas.version}";
+      in ''
+        XDG_DATA_DIRS=$XDG_DATA_DIRS:${gsettings-schemas} \
+        DBUS_SESSION_BUS_ADDRESS="unix:path=''${XDG_RUNTIME_DIR:-/run/user/$UID}/bus" \
```

These variables should be available in the reload script.

Otherwise,
```suggestion
        XDG_DATA_DIRS="$XDG_DATA_DIRS"${XDG_DATA_DIRS:+:}${escapeShellArg gsettings-schemas} \
        DBUS_SESSION_BUS_ADDRESS="unix:path=''${XDG_RUNTIME_DIR:-/run/user/$UID}/bus" \
```

---

I think these need to be explicitly set when using hm as a module, thanks

---

This change seems to result in a syntax error:
```
error: syntax error, unexpected '+', at /etc/nixos/imports/home-manager/modules/misc/gtk.nix:188:55
(use '--show-trace' to show detailed location information)
```